pkg/utils/config/main.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	tomlv2 "github.com/pelletier/go-toml/v2"
+)
+
+const (
+	tomlFilePath = "/etc/containerd/config.toml"
+	nriPluginKey = "io.containerd.nri.v1.nri"
+	disableKey   = "disable"
+)
+
+func main() {
+	// Load TOML content
+	tomlData, err := os.ReadFile(tomlFilePath)
+	if err != nil {
+		log.Fatalf("Error reading TOML file: %v", err)
+	}
+
+	tomlMap, err := unmarshalTOML(tomlData)
+	if err != nil {
+		log.Fatalf("Error unmarshalling TOML: %v", err)
+	}
+
+	tomlMap = updateNRIplugin(tomlMap)
+
+	buf, err := addIndentation(tomlMap)
+	if err != nil {
+		log.Fatalf("Error adding indentation: %v", err)
+	}
+
+	if err := writeToFile(tomlFilePath, buf.String()); err != nil {
+		log.Fatalf("Error writing to file: %v", err)
+	}
```

Comment:
Maybe it's just me but I'd
- combine reading and decoding to a single `readConfig(file string) (map[string]interface{}, error)` function (probably using just toml.NewDecoder().Decode()`
- have an `enableNRIPlugin(map [string]interface{}) error function (your `updateNRIplugin()` with fixes)
- combine encoding with indentation and file writing into a single `writeConfig(file string, config map[string]interface{}) error` which would just do a `toml.NewEncoder(f).SetIndentTables(true).Encode(config)`

Reply:
And this was merely just me thinking out loud. These changes are not required for merging this PR.